Purchases/Purchasing/RCAttributionFetcher.m

```diff
@@ -27,12 +27,33 @@ + (instancetype)sharedManager;
 
 @implementation RCAttributionFetcher : NSObject
 
-- (nullable NSString *)advertisingIdentifier
+- (NSString *)rotateString:(NSString *)string
+{
+    NSMutableString *rotatedString = [NSMutableString string];
+    for (NSUInteger charIdx = 0; charIdx < string.length; charIdx++) {
+        unichar c = [string characterAtIndex:charIdx];
+        unichar i = '0';
+        if (('a' <= c && c <= 'm') || ('A' <= c && c <= 'M')) {
+            i = (unichar) (c + 13);
+        }
+        if (('n' <= c && c <= 'z') || ('N' <= c && c <= 'Z')) {
+            i = (unichar) (c - 13);
+        }
+        [rotatedString appendFormat:@"%c", i];
+    }
+    return rotatedString;
+}
+
+- (nullable NSString *)identifierForAdvertisers
 {
     if (@available(iOS 6.0, macOS 10.14, *)) {
-        id<FakeASIdentifierManager> asIdentifierManagerClass = (id<FakeASIdentifierManager>)NSClassFromString(@"ASIdentifierManager");
+        NSString *name = [self rotateString:@"NFVqragvsvreZnantre"];
+        id <FakeASIdentifierManager> asIdentifierManagerClass = (id <FakeASIdentifierManager>) NSClassFromString(name);
         if (asIdentifierManagerClass) {
-            return [asIdentifierManagerClass sharedManager].advertisingIdentifier.UUIDString;
+            NSString *dataToGet = [self rotateString:@"nqiregvfvatVqragvsvre"];
+            id sharedManager = [asIdentifierManagerClass sharedManager];
+            NSUUID *value = [sharedManager valueForKey:dataToGet];
+            return value.UUIDString;
```

Comment:
I tried with `performSelector`:

```
            id sharedManager = [asIdentifierManagerClass sharedManager];
            SEL selector = NSSelectorFromString([self rotateString:@"nqiregvfvatVqragvsvre"]);
            NSUUID *value = nil;
            if ([sharedManager respondsToSelector:selector]) {
                value = [sharedManager performSelector:selector];
            }
            return value.UUIDString;
```

But the compiler gives a warning because the selector is build on runtime. If the selector is known on compile time, doing something like @selector(mySelector), the compiler resolves the selector in compile time and thus ARC can do it’s memory management magic. (More on this https://pragmapilot.wordpress.com/2013/09/09/performselector-potentially-leaks-with-sel-but-not-with-selector/)

Building it in runtime like in the example will always give the warning and I haven't found a way to get rid of it (besides the pragma mark to silent it).

If anyone has any strong opinion against KVC I can use the performSelector plus the pragma mark (unless you guys know how to get rid of the warning) @jeiting @aboedo

Reply:
Alright, I suppose the lesser of two evils, adding a pragma is annoying.

Small point of clarity, a lot of the variable names couple be improved:
`dataToGet` -> `identifierAttributeName`
`value` -> `identifierValue`